async-coap-uri/src/error.rs

```diff
@@ -61,6 +80,75 @@ impl ParseError {
 
     /// A debugging description of the error.
     pub fn desc(&self) -> &'static str {
-        self.desc
+        self.desc.as_str()
+    }
+}
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.desc)
+    }
+}
+
+#[cfg(feature = "std")]
+impl ::std::error::Error for ParseError {}
+
+impl From<crate::escape::DecodingError> for ParseError {
+    fn from(error: crate::escape::DecodingError) -> Self {
+        Self {
+            span: Some(error.index..error.index + 1),
+            desc: ParseErrorKind::EncodingError(error),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+enum ParseErrorKind {
+    /// Bad percent encoding or illegal characters
+    EncodingError(crate::escape::DecodingError),
```

Comment:
I get why this is named `EncodingError`, but it does feel a little weird that the variant named `EncodingError` contains a single variable with a type named `DecodingError`.